controllers/kustomization_impersonation.go

```diff
@@ -195,8 +195,15 @@ func (ki *KustomizeImpersonation) getKubeConfig(ctx context.Context) ([]byte, er
 		return nil, fmt.Errorf("unable to read KubeConfig secret '%s' error: %w", secretName.String(), err)
 	}
 
-	kubeConfig, ok := secret.Data["value"]
-	if !ok {
+	var kubeConfig []byte
+	for k, _ := range secret.Data {
+		if k == "value" || k == "value.yaml" {
```

Comment:
Given these are parts of a contract (they are defined explicitly in the spec), I would make them more "official" part of the programmatic API by e.g. making them constants.

Reply:
This must move to `fluxcd/pkg/runtime/client` so I would leave it here as it is since this function will no longer be in each controller.